sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/LogicalTypes.java

```diff
@@ -266,8 +266,8 @@ private FixedPrecisionNumeric(
     @Override
     public BigDecimal toInputType(BigDecimal base) {
       checkArgument(
-          base == null || (base.precision() == precision && base.scale() == scale),
-          "Expected BigDecimal base to be null or have precision = %s (was %s), scale = %s (was %s)",
+          base == null || (base.precision() <= precision && base.scale() <= scale),
```

Comment:
Incidentally it took me a minute to understand what is happening here because I don't know why it is called `base`. This is converting the value returned by JDBC to the value expected by Beam schema, right?

I am having trouble wrapping my head around all the cases here, or coming up with the simple math. The inefficient but simple check here is `base.round(new MathContext(precision)).compareTo(base) == 0`.

If the scales are equal, then lower precision is fine, certainly.

But you could even have "1000 x 10^-3" (precision 3, scale 3) and it can safely coerce to "1 x 10^0" (precision 1, scale 0). Am I getting this wrong?

Reply:
@kennknowles would

`(base.precision() - base.scale()) <= (precision - scale) && (base.scale() <= scale)`
work?

Reply:
@borovikovd Looks like it doesn't work with @kennknowles's example because `base.scale()` (3) > `scale` (0).

Reply:
I think we can add smth like
```java
base == null || (base.precision() <= precision && base.scale() <= scale)
              // for cases when received values can be safely coerced to the schema 
              || base.round(new MathContext(precision)).compareTo(base) == 0
```
I suppose, the first condition will be enough for the vast majority of cases, when the second one will be executed for negative cases and cases when values can be safely coerced to the schema.